lib/cocoapods-core/specification/dsl.rb

```diff
@@ -677,8 +677,21 @@ def dependency(*args)
           end
         end
         unless version_requirements.all? { |req| req.is_a?(String) }
-          raise Informative, 'Unsupported version requirements'
+          version_requirements.each do |requirement|
+            if requirement.is_a?(Hash)
+              if !requirement[:path].nil?
+                raise Informative, 'Podspecs cannot specify the source of dependencies. The :path option is not supported.'\
```

Comment:
nit can you wrap :path with backticks \`\`? Same for :git below

Reply:
done